datasource/opennms-direct/src/main/java/org/opennms/alec/datasource/opennms/jvm/ApiMapper.java

```diff
@@ -76,6 +76,7 @@ public Alarm toAlarm(org.opennms.integration.api.v1.model.Alarm alarm) {
         ImmutableAlarm.Builder alarmBuilder = ImmutableAlarm.newBuilder();
         alarmBuilder
                 .setId(alarm.getReductionKey())
+                .setFirstTime(alarm.getFirstEventTime() != null ? alarm.getFirstEventTime().getTime() : 0l)
```

Comment:
```suggestion
                .setFirstTime(alarm.getFirstEventTime().getTime())
```

should always be non-null